central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -630,6 +720,41 @@ func (s *flowStoreImpl) pruneFlows(ctx context.Context, deleteStmt string, orpha
 	return nil
 }
 
+func (s *flowStoreImpl) pruneAndReturnFlows(ctx context.Context, deleteStmt string, orphanWindow *time.Time) ([]*storage.NetworkFlow, error) {
+	conn, release, err := s.acquireConn(ctx, ops.Remove, "NetworkFlow")
```

Comment:
Do you need to get this `conn` every time? Maybe make it a member of `flowStroreImpl`.

Reply:
One important part of those initialization incantations are the `defer()` calls for the connection and the context. AFAIK, defer() calls cannot be delegated to a subfunction.

Do you see a way to delegate the `defer()` calls ?

Reply:
IMO if this is something we wanted to do, it would probably be better to grab the connection once in `RemoveOrphanedFlows` and pass it into each helper function.  There is probably value in looking at that.

I agree with @ovalenti that creating a subfunction to get the connection itself is likely not correct.  And I don't think we could add it to the type because we would have to manage opening and closing.

Reply:
Looking at this further, I am not sure that it is worth it to make this change here.